power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -714,50 +766,113 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
             // Remember the begin iterator of the first scenario, then loop over the remaining scenarios and check the
             // ids
             auto const first_span = all_spans[0];
             // check the subsequent scenarios
             // only return true if all scenarios match the ids of the first batch
-            return std::all_of(all_spans.cbegin() + 1, all_spans.cend(), [&first_span](auto const& current_span) {
-                return std::ranges::equal(
-                    current_span, first_span,
-                    [](UpdateType<CT> const& obj, UpdateType<CT> const& first) { return obj.id == first.id; });
-            });
+            result.ids_match =
+                std::all_of(all_spans.cbegin() + 1, all_spans.cend(), [&first_span](auto const& current_span) {
+                    return std::ranges::equal(
+                        current_span, first_span,
+                        [](UpdateType<CT> const& obj, UpdateType<CT> const& first) { return obj.id == first.id; });
+                });
         };
 
-        auto const is_component_update_independent = [&update_data, &process_buffer_span]<typename CT>() -> bool {
+        auto const check_each_component = [&update_data, &process_buffer_span]<typename CT>() -> UpdateCompProperties {
             // get span of all the update data
-            if (update_data.is_columnar(CT::name)) {
-                return process_buffer_span.template operator()<CT>(
-                    update_data.get_columnar_buffer_span_all_scenarios<meta_data::update_getter_s, CT>());
+            auto const comp_index = update_data.find_component(CT::name, false);
+            UpdateCompProperties result;
+            result.name = CT::name;
+            result.is_columnar = update_data.is_columnar(result.name);
+            if (comp_index >= 0) {
+                result.elements_ps_in_update = update_data.get_component_info(comp_index).elements_per_scenario;
+            }
+            if (result.is_columnar) {
+                process_buffer_span.template operator()<CT>(
+                    update_data.get_columnar_buffer_span_all_scenarios<meta_data::update_getter_s, CT>(), result);
+            } else {
+                process_buffer_span.template operator()<CT>(
+                    update_data.get_buffer_span_all_scenarios<meta_data::update_getter_s, CT>(), result);
             }
-            return process_buffer_span.template operator()<CT>(
-                update_data.get_buffer_span_all_scenarios<meta_data::update_getter_s, CT>());
+            return result;
         };
 
-        // check all components
-        auto const update_independent = run_functor_with_all_types_return_array(is_component_update_independent);
-        return std::ranges::all_of(update_independent, [](bool const is_independent) { return is_independent; });
+        // check and return indenpendence of all components
+        return run_functor_with_all_types_return_vector(check_each_component);
```

Comment:
Just curious, maybe I missed something, but why call here (and iin other places), the `run_functor*` function instead of directly the lambda?

Reply:
The context requires as return a list of result from the application of functors on each component type.